File: task22/main.py
def shaker_sort(arr):
    arr = arr[:]
    left, right = 0, len(arr) - 1
    while left < right:
        for i in range(left, right):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
        right -= 1
        for i in range(right, left, -1):
            if arr[i] < arr[i - 1]:
                arr[i], arr[i - 1] = arr[i - 1], arr[i]
        left += 1
    return arr

if __name__ == '__main__':
    # Ручной ввод массива
    raw = input("Введите числа через пробел: ")
    arr = list(map(int, raw.split()))

    # Вариант чтения из файла:
    # with open('input.txt', 'r') as f:
    #     arr = list(map(int, f.read().split()))

    print("До:    ", arr)
    print("После: ", shaker_sort(arr))

File: task22/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <sstream>
#include <string>

void shakerSort(std::vector<int>& arr) {
    int left = 0, right = (int)arr.size() - 1;
    while (left < right) {
        for (int i = left; i < right; ++i)
            if (arr[i] > arr[i + 1])
                std::swap(arr[i], arr[i + 1]);
        --right;

        for (int i = right; i > left; --i)
            if (arr[i] < arr[i - 1])
                std::swap(arr[i], arr[i - 1]);
        ++left;
    }
}

int main() {
    // Ручной ввод массива
    std::cout << "Введите числа через пробел: ";
    std::string line;
    std::getline(std::cin, line);
    std::istringstream iss(line);
    std::vector<int> arr;
    int x;
    while (iss >> x) arr.push_back(x);

    // Вариант чтения из файла:
    // #include <fstream>
    // std::ifstream fin("input.txt");
    // std::vector<int> arr;
    // int x;
    // while (fin >> x) arr.push_back(x);

    std::cout << "До:    ";
    for (int v : arr) std::cout << v << ' ';
    std::cout << '\n';

    shakerSort(arr);

    std::cout << "После: ";
    for (int v : arr) std::cout << v << ' ';
    std::cout << '\n';
    return 0;
}